internal/compress/lz4.go

```diff
@@ -121,10 +132,13 @@ type lz4Writer struct {
 	w   io.WriteCloser
 }
 
+// Write returns the number of bytes written from p (0 <= n <= len(p)).
+// If any errors occurs, it will return an error.
 func (l *lz4Writer) Write(p []byte) (n int, err error) {
 	return l.w.Write(p)
 }
 
+// Close closes dst and w.
 func (l *lz4Writer) Close() (err error) {
 	err = l.w.Close()
 	if err != nil {
```

Comment:
I am not sure should we return err here. If error occurs the `dst` will not be closed.

Reply:
Both of `dst` and `w` returns error.

Reply:
I mean in line 145, if `l.w.Close()` returns error, line 148 will not executed and I am not sure if it is correct implementation.

Reply:
```return errors.Wrap(l.dst.Close(), err.Error())```
It seems no problem. 🤔 
error of `l.w.Close()`will be wrapped with error of `l.dst.Close()`.